src/Entity/Geo/Canton.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Entity\Geo;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(
+ *     name="geo_canton",
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="geo_canton_code_unique", columns="code")
+ *     }
+ * )
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class Canton
+{
+    use GeoTrait;
+
+    /**
+     * @var Department|null
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Geo\Department")
+     * @ORM\JoinColumn(nullable=false)
```

Comment:
if the relatoin is `nullable=false`:
- `@var Department|null` => `@var Department`
- we should have `__construct(Department $department)`
- `getDepartment(): ?Department` => `getDepartment(): Department`

(same for other properties in the same case)

Reply:
for all classes that belongs to department (1) I created a `UnderDepartmentTrait` to handle this attribute and its getter and setter, and (2) declared it in constructor